integr8ly/cloud-resource-operator
# 🐛 If resource marked as created, don't try create

## Overview

JIRA: https://issues.redhat.com/browse/INTLY-5819

If a resource was created on a cloud provider, but later was
accidentally deleted (or appears non-existent for some other reason),
the operator shouldn't try to create it again: manual intervention may
be needed (respond to alert, then restore from backup or delete CR).

This flag has been added as an annotation rather than a field in the
status block so that it can be manipulated manually. There are a
couple of cases where a human may want to do this:

- Restore an accidentally deleted CR that has/had a backing resource
on a cloud provider
- Migrate CR to a new cluster
- Force operator to create a new resource if restoration from backup
is not desirable

## Verification (updated with AWS-specific info)

Deploy the operator from this branch & configure to be able to create resources on AWS.

- Create one of each of Redis/Postgres/BlobStorage on AWS and make sure they all get created as expected, and then the "resourceIdentifier" annotation is set
- Maybe try deleting the backing resource for each of those CRs, and make sure that the operator doesn't recreate it.

- Create a CR for each of those CRDs with the annotation preset, and notice that the operator won't try to create anything for it.
